crates/kitsune_p2p/types/src/bloom_endpoint.rs

```diff
@@ -0,0 +1,29 @@
+use crate::{bin_types::*, *};
+
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, derive_more::Constructor)]
+pub struct BloomEndpoint {
+    agent: KitsuneAgent,
+    cert: Tx2Cert,
```

Comment:
Alas, I haven't settled on whether it's better to Arc inside a type or outside it... `Tx2Cert` has an internal arc, so it's safe to use it directly and trust your clones are cheap... KitsuneAgent however, is mostly referred to within kitsune as `Arc<KitsuneAgent>` because it does not have an internal arc, so cloning it is a waste of memory...